aoc/day_4.py

```diff
@@ -0,0 +1,39 @@
+import os 
+
+DATA_PATH="inputs/day_4.txt"
+
+def get_assignment_lines():
+    with open(DATA_PATH, 'r') as file:
+        return file.read().splitlines()
+
+def get_sections_list(assignment_row):
+    assignments = assignment_row.split(',')
+    full_assignment = []
+    for assignment in assignments:
+        start = int(assignment.split('-')[0])
+        stop = int(assignment.split('-')[1]) + 1
+        full_assignment.append(list(range(start, stop)))
```

Comment:
I would use a set here, not a list. We're comparing whether something exists in the set, not counting how many times it exists. Then "partial overlap" becomes "non-empty intersection" and "full overlap" becomes "subset"